web/elm/src/TopBar/TopBar.elm

```diff
@@ -466,11 +415,11 @@ viewLogin userState model isPaused =
         []
 
 
-showLogin : Model -> Bool
+showLogin : Model r -> Bool
 showLogin model =
-    case model.middleSection of
-        SearchBar _ ->
-            model.screenSize /= Mobile
+    case ( model.screenSize, middleSection model ) of
```

Comment:
Now that ScreenSize has no arguments, we can replace this with `showLogin model = model.middleSection /= SearchBar || model.screenSize /= mobile`. I think that's marginally clearer.